lib/_stream_readable.js

```diff
@@ -1070,13 +1072,84 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
-  if (createReadableStreamAsyncIterator === undefined) {
-    createReadableStreamAsyncIterator =
-      require('internal/streams/async_iterator');
+  let stream = this;
+
+  if (typeof stream.read !== 'function') {
+    // v1 stream
+    const src = stream;
+    stream = new Readable({
+      objectMode: true,
+      destroy(err, callback) {
+        destroyImpl.destroyer(src, err);
+        callback();
+      }
+    }).wrap(src);
   }
-  return createReadableStreamAsyncIterator(this);
+
+  const iter = createAsyncIterator(stream);
+  iter.stream = stream;
+  return iter;
 };
 
+async function* createAsyncIterator(stream) {
+  const state = stream._readableState;
+
+  if ((state && state.endEmitted) ||
+      (state && state.closeEmitted)) {
+    // TODO (ronag): What if errorEmitted?
+    // TODO (ronag): Wait for closeEmitted?
+    // TODO (ronag): if (closeEmitted && !endEmitted) ERR_PREMATURE_CLOSE?
+    stream.on('error', nop);
```

Comment:
Would you mind explaining why we're doing this?

I am not sure what the correct behavior here is or when that (closeEmitted or endEmitted being set but error being emitted at some future point) should happen.

Reply:
So in the case where end or close has not been emitted we always leave an error handler, which would mute any unhandled error. Ditto with the old implementation and finished/eos.

Reply:
This is consistent with behavior in pipeline, finished and old async iterator.

Reply:
actually, there might be an issue here since we are not destroying the stream in this case.

Reply:
I am not sure conceptually this behavior is correct. 

Why would getting an iterator for a stream implicitly destroy it if it is closed? Wouldn't it be destroyed beforehand if it is closed?